### PR TITLE
feat(frontend): support background ddl for materialized views

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Part of #8051 

Support running background ddl.
```
CREATE TABLE t(v1 int);
insert into t select * from generate_series(1, 1000000);
SET BACKGROUND_DDL=true;
-- Should return immediately:
CREATE MATERIALIZED VIEW m as select * from t;
```

We can support it without persistence first.

I prefer to keep this PR small, and work towards an e2e working background ddl for mviews first. Later we will handle sinks and indexes as well. And separately for persistence: https://github.com/risingwavelabs/risingwave/pull/12167

### Implementation details

- Extend `CreateMaterializedViewRequest` to take `background_ddl` as a parameter.
- On the meta side, we run prepare + build stream job first.
- Then to avoid blocking the response to frontend,
   we spawn the stream job creation task in a tokio thread.
- If there's any errors during creation we just emit meta logs for it.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

Users can now set the session variable `BACKGROUND_DDL` to a boolean value. If `true`, `CREATE MATERIALIZED VIEW` will execute in the background, and users should expect to immediately get the typical `CREATE MATERIALIZED VIEW` back as a response.

If `false`, `CREATE MATERIALIZED VIEW` will execute as per normal.

Eventually `CREATE SINK` and `CREATE INDEX` will support this as well.
Subsequent PRs for these will include a release note.

As an example:
```sql
CREATE TABLE t(v1 int);
insert into t select * from generate_series(1, 1000000);
flush;
SET BACKGROUND_DDL=true;
-- Should return immediately:
CREATE MATERIALIZED VIEW m as select * from t;
```

You can use `SHOW JOBS` to monitor its progress:
```sql
-- Call show jobs, may have a small delay before progress shows up.
-- This will display the progress of creating the materialized view m.
dev=> show jobs;
  Id  |                   Statement                   | Progress 
------+-----------------------------------------------+----------
 1002 | CREATE MATERIALIZED VIEW m AS SELECT * FROM t | 24.68%
(1 row)
```

Jobs running in the background can be cancelled:
```sql
dev=> cancel jobs 1002;
  Id  
------
 1002
(1 row)

dev=> show jobs;
 Id | Statement | Progress 
----+-----------+----------
(0 rows)
```

</details>

### CODE REVIEW

File: src/meta/src/rpc/service/ddl_service.rs [hunk 1]
```diff
@@ -229,7 +229,11 @@ impl DdlService for DdlServiceImpl {
 
         let version = self
             .ddl_controller
-            .run_command(DdlCommand::CreateStreamingJob(stream_job, fragment_graph))
+            .run_command(DdlCommand::CreateStreamingJob(
+                stream_job,
+                fragment_graph,
+                false,
```
Comment (chenzl25) [inline src/meta/src/rpc/service/ddl_service.rs:235]: Creating a sink is also a DDL job. I think we should pass `run_in_background` to this method as well.

File: src/meta/src/rpc/service/ddl_service.rs [hunk 2]
```diff
@@ -331,7 +339,11 @@ impl DdlService for DdlServiceImpl {
...
```
Comment (chenzl25) [inline src/meta/src/rpc/service/ddl_service.rs:345]: Creating an index is also a DDL job. I think we should pass `run_in_background` to this method as well.

File: src/meta/src/rpc/service/ddl_service.rs (hunk 1 above)
Comment (kwannoel) [reply src/meta/src/rpc/service/ddl_service.rs:235]: Actually was planning to add it in separate PR, as mentioned in PR description:
> Later we will handle sinks and indexes as well. 

But I'm okay to add within this PR too.

**Review by chenzl25 — APPROVED** [review]
LGTM！

**Review by yezizp2012 — APPROVED** [review]
LGTM! 
One thing non-related to this PR about recovery, as you said in slack channel: "If user does not set background ddl session variable, we don’t persist backfill", if so then we need a way to distinguish between jobs that need to be recovered and jobs that need to be cleaned.

File: e2e_test/ddl/background.slt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+DROP MATERIALIZED VIEW m1;
+
+statement ok
+DROP MATERIALIZED VIEW m2;
+
+statement ok
+DROP MATERIALIZED VIEW m3;
+
+statement ok
+DROP TABLE t;
```
Comment (yezizp2012) [inline e2e_test/ddl/background.slt:65]: Oh, I think you need to reset `SET BACKGROUND_DDL=false;` at the end of this test, so that it won't affect other tests. The failure of the tests in CI that non-related to recovery should be caused by this.

File: ci/workflows/pull-request.yml [hunk 4]
```diff
@@ -460,3 +460,19 @@ steps:
           mount-buildkite-agent: true
       - ./ci/plugins/upload-failure-logs
     timeout_in_minutes: 5
+
+  - label: "e2e standalone binary test"
```
Comment (BugenZhao) [inline ci/workflows/pull-request.yml:464]: Is this related to this PR?

File: src/meta/src/rpc/ddl_controller.rs [hunk 5]
```diff
@@ -81,6 +82,7 @@ impl StreamingJobId {
     }
 }
 
+pub type RunInBackground = bool;
```
Comment (BugenZhao) [inline src/meta/src/rpc/ddl_controller.rs:85]: I'd suggest making this an `enum`. `Foreground` and `Background` could be much more readable than `true` and `false`.

**Review by BugenZhao — APPROVED** [review]
May mark this PR as "user-facing" and write some release notes for the doc team. :)

File: src/meta/src/rpc/ddl_controller.rs (hunk 5 above)
Comment (kwannoel) [reply src/meta/src/rpc/ddl_controller.rs:85]: Changing it to `DdlExecutionMode` with the two modes you mentioned.

Open to better names.

**Comment by kwannoel** [comment]
The `background_ddl` test is only run for e2e test workflow.

Don't quite get why the other tests are failing as well, since no `BACKGROUND_DDL` variable is set. Investigating.

**Comment by kwannoel** [comment]
Test failure caused by failing to cleanup stream job on failure. Need to clean up the `build_stream_job` step as well.
